File: LeetCode/1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.java
class Solution {
    public int binarySearch(int[] mat, int n) {
        int low = 0, high = n - 1;
        while (low <= high) {
            int mid = low + (high - low) / 2;
            if (mat[mid] < 0) {
                high = mid - 1;
            } else {
                low = mid + 1;
            }
        }
        return n - low;
    }
    public int countNegatives(int[][] grid) {
        int row = grid.length, col = grid[0].length;
        int totalCount = 0;
        for (int i = 0; i < row; i++) {
            totalCount += binarySearch(grid[i], col);
        }
        return totalCount;
    }
}

File: LeetCode/1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.py
class Solution:
    def binarySearch(self, mat: List[int], n: int) -> int:
        low = 0
        high = n - 1
        while low <= high:
            mid = low + (high - low) // 2
            if mat[mid] < 0:
                high = mid - 1
            else:
                low = mid + 1
        return n - low

    def countNegatives(self, grid: List[List[int]]) -> int:
        row = len(grid)
        col = len(grid[0])
        totalCount = 0
        for i in range(row):
            totalCount += self.binarySearch(grid[i], col)
        return totalCount